src/opm/input/eclipse/EclipseState/Tables/TableManager.cpp

```diff
@@ -181,9 +181,28 @@ std::optional<JFunc> make_jfunc(const Deck& deck) {
 
         if( deck.hasKeyword( "DIFFCWAT" ) )
             this->m_diffCoeffWatTable = DiffCoeffWatTable( deck["DIFFCWAT"].back() );
+        else if (deck.hasKeyword( "DIFFAWAT" ) )
+            this->m_diffCoeffWatTable = DiffCoeffWatTable( deck["DIFFAWAT"].back() );
 
         if( deck.hasKeyword( "DIFFCGAS" ) )
             this->m_diffCoeffGasTable = DiffCoeffGasTable( deck["DIFFCGAS"].back() );
+        else if ( deck.hasKeyword( "DIFFAGAS" ) )
+            this->m_diffCoeffGasTable = DiffCoeffGasTable( deck["DIFFAGAS"].back() );
+        
+        if (deck.hasKeyword( "DIFFCGAS" ) && deck.hasKeyword( "DIFFAGAS" ))
+            throw OpmInputError("The DIFFCGAS and DIFFAGAS tables are mutually exclusive.", deck.get<ParserKeywords::DIFFCGAS>().back().location(), deck.get<ParserKeywords::DIFFAGAS>().back().location());
+
+        if (deck.hasKeyword( "DIFFCWAT" ) && deck.hasKeyword( "DIFFAWAT" ))
+            throw OpmInputError("The DIFFCWAT and DIFFAWAT tables are mutually exclusive.", deck.get<ParserKeywords::DIFFCWAT>().back().location(), deck.get<ParserKeywords::DIFFAWAT>().back().location());
+
+        if (deck.hasKeyword( "DIFFCGAS" ) && deck.hasKeyword( "DIFFAWAT" ))
+            throw OpmInputError("The DIFFCGAS and DIFFAWAT tables are mutually exclusive.", deck.get<ParserKeywords::DIFFCGAS>().back().location(), deck.get<ParserKeywords::DIFFAWAT>().back().location());
+
+        if (deck.hasKeyword( "DIFFCWAT" ) && deck.hasKeyword( "DIFFAGAS" ))
+            throw OpmInputError("The DIFFCWAT and DIFFAGAS tables are mutually exclusive.", deck.get<ParserKeywords::DIFFCWAT>().back().location(), deck.get<ParserKeywords::DIFFAGAS>().back().location());
```

Comment:
These would be easier to implement using the `requires`/`prohibits` mechanism of the JSON schema.  One example usage is in [`SWOF`](https://github.com/OPM/opm-common/blob/1224f8c974483a7c23bdfc844906461382d7031a/src/opm/input/eclipse/share/keywords/000_Eclipse100/S/SWOF#L10-L11).

Reply:
Thanks. Will update.